contracts/rust/src/bin/diff_test.rs

```diff
@@ -387,6 +393,26 @@ fn main() {
         Action::TestOnly => {
             println!("args: {:?}", cli.args);
         }
+        Action::GenBLSSig => {
+            let mut rng = jf_utils::test_rng();
+
+            let key_pair = KeyPair::generate(&mut rng);
+            let vk = key_pair.ver_key();
+            let vk_g2_proj: G2Affine = vk.to_affine();
+            let msg = Bytes::from(vec![54u8, 12u8]);
+            let pk_x_c0 = field_to_u256::<Fq>(vk_g2_proj.x.c0);
+            let pk_x_c1 = field_to_u256::<Fq>(vk_g2_proj.x.c1);
+            let pk_y_c0 = field_to_u256::<Fq>(vk_g2_proj.y.c0);
+            let pk_y_c1 = field_to_u256::<Fq>(vk_g2_proj.y.c1);
+
+            let sig: Signature = key_pair.sign(&msg, CS_ID_BLS_BN254);
+            let sig_affine_point = sig.sigma.into_affine();
+            let sig_x = field_to_u256::<Fq>(sig_affine_point.x);
+            let sig_y = field_to_u256::<Fq>(sig_affine_point.y);
+
+            let res = (sig_x, sig_y, pk_x_c0, pk_x_c1, pk_y_c0, pk_y_c1, msg);
+            println!("{}", res.encode_hex());
```

Comment:
We should define a `struct ParsedG2Point` (instead of parsing each field individually which is more manual and error prone), and implement two-way conversion, see `ParsedG1Point` for example.

Reply:
TODO created in b0ba4b6222fc9c02e9c65111c9663ebff4400d09